src/data_info/services/trash.clj

```diff
@@ -56,31 +55,36 @@
    (irods/with-jargon-exceptions [cm]
      (delete-paths cm user paths)))
   ([cm user paths]
-     (let [paths (mapv ft/rm-last-slash paths)
-           trash-paths (atom (hash-map))]
+     (let [paths (mapv ft/rm-last-slash paths)]
        (validators/user-exists cm user)
        (validators/all-paths-exist cm paths)
        (validators/user-owns-paths cm user paths)
        (validate-not-homedir user paths)
 
-       (doseq [^String p paths]
-         (log/debug "path" p)
-         (log/debug "readable?" user (owns? cm user p))
-
-         ;;; Delete all of the tickets associated with the file.
-         (let [path-tickets (mapv :ticket-id (ticket-ids-for-path cm (:username cm) p))]
-           (doseq [path-ticket path-tickets]
-             (delete-ticket cm (:username cm) path-ticket)))
-
-         ;;; If the file isn't already in the user's trash, move it there
-         ;;; otherwise, do a hard delete.
-         (if-not (.startsWith p (paths/user-trash-path user))
-           (do (let [trash-path (move-to-trash cm p user)]
-               (swap! trash-paths assoc p trash-path)))
-           (delete cm p true))) ;;; Force a delete to bypass proxy user's trash.
-
-        {:paths paths
-         :trash-paths @trash-paths})))
+       (let [trash-paths (apply merge (mapv
+                           (fn [path]
+                             (if-not (.startsWith path (paths/user-trash-path user))
+                               {path (randomized-trash-path user path)}
+                               {}))
+                           paths))]
```

Comment:
This `let` is the main point here, it precalculates the locations to move things into in the trash, such that the `doseq` below could ostensibly be moved to an async thread but still allow us to return the trash-paths map as we have been.